### PR TITLE
Consistency with GPflux/GPflow reparameterization samplers

### PR DESCRIPTION
**Pull request by sebastianober**
Ensuring consistency with shapes between MCEI/MCAE and GPflux/GPflow reparameterization samplers.

### CODE REVIEW

File: tests/integration/test_bayesian_optimization.py [hunk 1]
```diff
@@ -96,6 +99,22 @@ def GPR_OPTIMIZER_PARAMS() -> Tuple[str, List[ParameterSet]]:
...
+                id="MonteCarloExpectedImprovement",
+            ),
+            pytest.param(
+                20,
+                EfficientGlobalOptimization(
+                    MonteCarloAugmentedExpectedImprovement(int(1e3)).using(OBJECTIVE),
+                    generate_continuous_optimizer(100),
+                ),
+                id="MonteCarloAugmentedExpectedImprovement",
+            ),
```
Comment (henrymoss) [inline tests/integration/test_bayesian_optimization.py]: Perhaps we dont need a full test for the augmented one?

File: tests/integration/test_bayesian_optimization.py [hunk 2]
```diff
@@ -325,7 +344,26 @@ def test_bayesian_optimizer_with_dgp_finds_minima_of_scaled_branin(
...
+            id="MonteCarloExpectedImprovement",
+        ),
+        pytest.param(
+            5,
+            EfficientGlobalOptimization(
+                MonteCarloAugmentedExpectedImprovement(int(1e2)), generate_continuous_optimizer(100)
+            ),
+            id="MonteCarloAugmentedExpectedImprovement",
+        ),
+    ],
```
Comment (henrymoss) [inline tests/integration/test_bayesian_optimization.py]: same comment as above

File: tests/unit/acquisition/function/test_function.py [hunk 3]
```diff
@@ -264,10 +264,13 @@ def test_mc_expected_improvement_builds_expected_improvement_using_best_from_mod
         tf.constant([[-2.0], [-1.0], [0.0], [1.0], [2.0]]),
         tf.constant([[4.1], [0.9], [0.1], [1.1], [3.9]]),
     )
+    # model = QuadraticMeanAndRBFKernel()
```
Comment (henrymoss) [inline tests/unit/acquisition/function/test_function.py]: eh?

File: tests/unit/acquisition/function/test_function.py [hunk 4]
```diff
@@ -264,10 +264,13 @@ def test_mc_expected_improvement_builds_expected_improvement_using_best_from_mod
...
     model = QuadraticMeanAndRBFKernelWithSamplers(dataset)
     acq_fn = MonteCarloExpectedImprovement(int(1e6)).prepare_acquisition_function(model, dataset)
     xs = tf.linspace([[-10.0]], [[10.0]], 100)
     expected = expected_improvement(model, tf.constant([0.0]))(xs)
+    print(expected.shape)
+    print(acq_fn(xs).shape)
```
Comment (henrymoss) [inline tests/unit/acquisition/function/test_function.py]: Quick delete this sloppy code before @uri-granta sees

File: tests/util/models/gpflow/models.py [hunk 5]
```diff
@@ -148,7 +148,7 @@ def covariance_between_points(
     def reparam_sampler(
         self: GaussianProcessWithoutNoise, num_samples: int
     ) -> ReparametrizationSampler[GaussianProcessWithoutNoise]:
-        return GaussianProcessSampler(num_samples, self)
+        return IndependentReparametrizationSampler(num_samples, self)
```
Comment (henrymoss) [inline tests/util/models/gpflow/models.py:152]: This is good. I was going to ask where this GaussianProcessSampler came from. Not sure it was ever needed

File: trieste/acquisition/function/function.py [hunk 6]
```diff
@@ -664,14 +666,16 @@ def update_acquisition_function(
...
+        )
+        mean = tf.reduce_mean(samples_at_query_points, axis=-3, keepdims=True)
 
         tf.debugging.assert_shapes(
-            [(mean, ["_", 1])], message="Expected model with output dimension [1]."
+            [(mean, [..., 1])], message="Expected model with output dimension [1]."
         )
 
-        eta = tf.reduce_min(mean, axis=0)
+        eta = tf.squeeze(tf.reduce_min(mean, axis=0))
```
Comment (henrymoss) [inline trieste/acquisition/function/function.py:678]: bit weird that it ends up with this extra dim?

**Review by henrymoss — COMMENTED** [review]
Looks good. I've only done a quick check so far though.

**Review by hstojic — COMMENTED** [review]
looks ok to me, but will leave @henrymoss  to approve

File: tests/integration/test_bayesian_optimization.py (hunk 1 above)
Comment (sebastianober) [reply tests/integration/test_bayesian_optimization.py]: I think we should have at least one just in case someone manages to break it, although you're right probably not both. Should we keep the slow test or the fast test?

File: tests/unit/acquisition/function/test_function.py (hunk 3 above)
Comment (sebastianober) [reply tests/unit/acquisition/function/test_function.py]: Whoops...

File: tests/unit/acquisition/function/test_function.py (hunk 4 above)
Comment (sebastianober) [reply tests/unit/acquisition/function/test_function.py]: Done ;)

File: tests/util/models/gpflow/models.py (hunk 5 above)
Comment (sebastianober) [reply tests/util/models/gpflow/models.py:152]: It's because the shapes my acquisition functions were expecting were different to the shapes `IndependentReparametrizationSampler` was giving, so I had to write my own utility `GaussianProcessSampler` to give the correct shapes. Probably not the best move in retrospect...

File: trieste/acquisition/function/function.py (hunk 6 above)
Comment (sebastianober) [reply trieste/acquisition/function/function.py:678]: So `mean` has shape `[N, 1, B, L]` where both `B` and `L` end up being 1, which should be consistent with the reparameterization code you already wrote for GPflow models. I can try to get rid of the extra dimensions elsewhere if that would be better?

File: tests/integration/test_bayesian_optimization.py (hunk 1 above)
Comment (hstojic) [reply tests/integration/test_bayesian_optimization.py]: keep the fast one I think

**Review by hstojic — APPROVED** [review]
Henry might need 2 more weeks since he has some  paper deadlines - to me it seems fine so I'll  approve this
